### PR TITLE
feat: [OIP-421]: Add baseline based verification 

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

image

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import css from '../TestSummaryView/TestsSummaryView.module.scss'
+
+interface PipelineBaselineProps {
+  data: VerificationOverview | null
+  activityId?: string
+}
+
+export default function PinBaslineButton(props: PipelineBaselineProps): JSX.Element {
+  const { data, activityId } = props
+
```
Comment (harshilbits) [inline src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx:21]: NIT: better to remove these empty lines.

File: src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          {getString('cancel')}
+        </Button>
+      </Dialog>
+    ),
+    [updatePinnedBaselineDetails]
+  )
+
+  const handlePinChange = useCallback(async () => {
+    const currentBaseline = data?.baselineOverview?.baselineVerificationJobInstanceId
+    if (currentBaseline && currentBaseline !== activityId && !isPinned) {
```
Comment (harshilbits) [inline src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx:79]: if condition can be moved to a function and unit tested.

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx [hunk 3]
```diff
@@ -174,28 +180,40 @@ export function Baseline(props: BaseFieldProps): JSX.Element {
...
-      <FormInput.MultiTypeInput
-        name={name ? name : 'baseline'}
-        style={style}
-        label={label ? label : getString('connectors.cdng.baseline')}
-        selectItems={getDefaultBaselineOptions(getString)}
-        multiTypeInputProps={getMultiTypeInputProps(expressions, allowableTypes)}
-      />
+      <Layout.Vertical>
+        <FormInput.MultiTypeInput
+          name={name ? name : 'baseline'}
```
Comment (harshilbits) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx:191]: can write as `name ?? 'baseline'`

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx [hunk 4]
```diff
@@ -174,28 +180,40 @@ export function Baseline(props: BaseFieldProps): JSX.Element {
...
+          style={style}
+          label={label ? label : getString('connectors.cdng.baseline')}
```
Comment (harshilbits) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx:193]: same : `label ?? getString('connectors.cdng.baseline')`

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx [hunk 5]
```diff
@@ -174,28 +180,40 @@ export function Baseline(props: BaseFieldProps): JSX.Element {
...
-        name={name ? name : 'baseline'}
-        style={style}
-        label={label ? label : getString('connectors.cdng.baseline')}
-        items={getDefaultBaselineOptions(getString)}
-        value={(formik?.values as ContinousVerificationData).spec?.spec?.baseline as SelectOption}
-        disabled={true}
-      />
+      <Layout.Vertical>
+        <FormInput.Select
+          name={name ? name : 'baseline'}
```
Comment (harshilbits) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx:206]: same.

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/constants.ts [hunk 6]
```diff
@@ -57,7 +57,10 @@ export const VerificationSensitivityOptions: SelectOption[] = [
   { label: 'Low', value: 'LOW' }
 ]
 
-export const baseLineOptions: SelectOption[] = [{ label: 'Last Successful job run', value: 'LAST' }]
+export const baseLineOptions: SelectOption[] = [
+  { label: 'Last Successful job run', value: 'LAST' },
+  { label: 'Pinned baseline', value: 'PINNED' }
```
Comment (harshilbits) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/constants.ts:62]: Define the label in strings.

Comment (deepesh-ui) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/constants.ts:62]: +1

Comment (praneshg239) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/constants.ts:62]: This is used in `ContinousVerificationStep` class where `getString` is not accessible. It is used for step registration. You have already used the same for `VerificationSensitivityOptions`, `trafficSplitPercentageOptions`, `durationOptions` etc,.

File: src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx]: Done

File: src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/components/ExecutionVerification/components/DeploymentProgressAndNodes/components/PinBaslineButton/PinBaslineButton.tsx]: Done

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx (hunk 3 above)
Comment (praneshg239) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx]: Done

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx (hunk 4 above)
Comment (praneshg239) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx]: Done

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx (hunk 5 above)
Comment (praneshg239) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.tsx]: Done

**Review by harshilbits — DISMISSED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**